bootstore/src/schemes/v0/peer.rs

```diff
@@ -0,0 +1,810 @@
+                Some(msg) = self.conn_rx.recv() => self.on_conn_msg(msg).await,
+                _ = interval.tick() => {
+                    if let Err(errors) = self.fsm.tick(Instant::now().into()) {
+                        for (_, err) in errors {
+                            self.handle_api_error(err).await;
+                        }
+                    }
+                    // Even with errors there may be messages that need sending
+                    self.deliver_envelopes().await;
+                }
+            }
+        }
+    }
+
+    // Handle an accepted connection
+    async fn on_accept(
+        &mut self,
+        res: Result<(TcpStream, SocketAddr), std::io::Error>,
+    ) {
+        match res {
+            Ok((sock, addr)) => {
+                let SocketAddr::V6(addr) = addr else {
+                    warn!(self.log, "Got connection from IPv4 address {addr}");
+                    return;
+                };
+                // TODO: Log if a peer with a lower address connects?
+                // Remove any existing connection
+                self.remove_accepted_connection(&addr).await;
+                info!(self.log, "Accepted connection from {addr}");
+                self.handle_unique_id_counter += 1;
+                let handle = spawn_accepted_connection_management_task(
+                    self.handle_unique_id_counter,
+                    self.config.id.clone(),
+                    self.config.addr.clone(),
+                    addr.clone(),
+                    sock,
+                    self.conn_tx.clone(),
+                    &self.log,
+                )
+                .await;
+                self.accepted_connections.insert(addr, handle);
+            }
+            Err(err) => {}
+        }
+    }
+
+    // Handle API requests from the `NodeHandle`
+    async fn on_api_request(&mut self, request: NodeApiRequest) {
+        match request {
+            NodeApiRequest::InitRack {
+                rack_uuid,
+                initial_membership,
+                responder,
+            } => {
+                info!(self.log,
+                    "Rack init started";
+                    "rack_uuid" => rack_uuid.to_string()
+                );
+                if self.init_responder.is_some() {
+                    let _ = responder
+                        .send(Err(NodeRequestError::RequestAlreadyPending));
+                    return;
+                }
+                if let Err(err) = self.fsm.init_rack(
+                    Instant::now().into(),
+                    rack_uuid,
+                    initial_membership,
+                ) {
+                    let _ = responder.send(Err(err.into()));
+                } else {
+                    self.init_responder = Some(responder);
+                    self.deliver_envelopes().await;
+                }
+            }
+            NodeApiRequest::InitLearner { responder } => {
+                if self.init_responder.is_some() {
```

Comment:
Add a log here similar to "Rack init started" above?

Reply:
Done in 94cf22c86